include/deal.II/base/quadrature_lib.h

```diff
@@ -439,6 +439,71 @@ class QSorted : public Quadrature<dim>
                   const std::pair<double, Point<dim> > &b);
 };
 
+/**
+ * Telles quadrature of arbitrary order.
+ *
+ * The coefficients of these quadrature rules are computed using
+ * a non linear variable change starting from a Gauss-Legendre
+ * quadrature formula.
+ * This is done using a cubic polynomial,
+ * n = a x^3 + b x^2 + c x + d
+ * in order to integrate
+ * a singular integral, with singularity at a given point x_0.
+ *
+ * We start from a Gauss Quadrature Formula with arbitrary
+ * function. Then we apply the cubic variable change.
+ * In his paper Telles applies the transformation on the reference
+ * cell [-1, 1] getting
+ *
+ * n(1) = 1
+ * n(-1) = -1
+ * dn/dx = 0 at x = x_0
+ * d2n/dx2 = 0 at x = x_0
+ *
+ * We get
+ * a = 1/q,
+ * b = -3gamma_bar/q,
+ * c = 3gamma_bar/q,
+ * d = -b,
+ * with
+ * eta_star=eta_bar^2 - 1;
+ * gamma_bar= nthroot( eta_bar*eta_star + abs(eta_star) ,3) + nthroot(eta_bar*eta_star - abs(eta_star),3)
+ * + eta_bar;
+ * q=((gamma-gamma_bar).^3 + gamma_bar*(gamma_bar^2+3))/(1+3*gamma_bar^2);
+ *
+ * Since the library assumes [0,1] as reference interval, we will map
+ * these values on the proper reference interval in the implementation.
+ *
+ * This variable change can be used to integrate singular integrals.
+ * One example is $\f(x)/abs(x-x_0)$ on the reference interval $[0,1]$,
+ * where $x_0$ is given at construction time, and is the location of the
+ * singularity $x_0$, and f(x) is a smooth non singular function.
+ *
+ * Singular quadrature formula are rather expensive, nevertheless Telles'
+ * quadrature formula are much easier to compute with respect to other singular
+ * integration techniques as Lachat-Watson.
+ *
+ * The weights and functions for Gauss Legendre formula have been tabulated up
+ * to order 12.
+ *
+ */
+template <int dim>
+class QTelles: public Quadrature<dim>
+{
+public:
+    /**
+      * In the constructor we can specify the order and the singular point,
+      * in this case we will map Gauss Legendre quadrature formulas following Telles'
+      * rule.
+    **/
+    QTelles (const unsigned int n, const Point<dim> singularity);
+    /**
+      * In the second constructor we can specify the quadrature formula to be mapped
+      * using Telles' rule, and the singular point.
+    **/
+
+    QTelles (const Quadrature<1> base_quad, const Point<dim> singularity);
```

Comment:
Can you change the order of the two constructors? What about the following descriptions?
"A constructor that takes a quadrature formula and a singular point as argument. The quadrature formula will be mapped using Telles' rule. Make sure that the order of the quadrature rule is appropriate for the singularity in question."

"A variant of above constructor that takes as parameters the order @p n and location of a singularity. A Gauss Legendre quadrature of order n will be used."